pkg/operator/ceph/version/version.go

```diff
@@ -257,7 +269,7 @@ func IsSuperior(a, b CephVersion) bool {
 	if a.Major == b.Major {
 		if a.Minor == b.Minor {
 			if a.Extra == b.Extra {
-				if a.Build > b.Build {
+				if a.Build > b.Build || (a.Build == b.Build && a.BuildSuffix != b.BuildSuffix) {
```

Comment:
same functionality, but to keep the structure similar to other checks:
```suggestion
				if a.Build > b.Build {
					return true
				}
				if a.Build == b.Build {
					if a.BuildSuffix != b.BuildSuffix) {
						return true
					}
				}			
```